catascript/catascript.py

```diff
@@ -0,0 +1,217 @@
+    
+    #Results
+    return catalog_files
+
+def load_TICS_dict():
+    """
+    This function loads the dictionnary of extracted TICS that have a ligth curve (it was extracted and stored by the extrattic module)
+
+    :returns: a dictionnary containing the (TIC, list of name_of_ligthcurve_file) pairs.
+    :rtype: dict
+    """
+    path_to_load = os.getenv("PATH_TO_EXTRACTED_TICS")
+    with open(path_to_load, 'rb') as pickle_dict:
+        dict_TICS = pickle.load(pickle_dict)
+    return dict_TICS
+
+#Preprocessing function
+def preprocess_catalog_line(catalog_line):
+    """
+    Preprocess the given catalog line into a dict with (name of field in the database, value for this line) entries.
+
+    :param catalog_line:
+    :type catalog_line: list
+
+    :returns: the processed line
+    :returns: dict
+    """
+    #Getting fields (in order)
+    list_of_fields = (os.getenv("LIST_DB_FIELDS")).split(',')
+
+    #Creating dict 
+    catalog_line_dict = {}
+
+    #Adding values, taking into account missing ones
+    for (index, field) in enumerate(list_of_fields):
+        if catalog_line[index] != '':
+            catalog_line_dict[field] = catalog_line[index]
+        else:
+            catalog_line_dict[field] = None
+
+    #results
+    return catalog_line_dict
+
+#Checking functions
+def check_in_TICS_dict(TIC_in_catalog, TICS_dict):
+    """
+    This function checks for the existence of a specified TIC in the dictionnary of TICS of which we have a ligth curve 
+    and returns the corresponding boolean.
+
+    :param TIC_in_catalog: the TIC we want to check
+    :type TIC_in_catalog: str
+    :param TICS_dict: the dict of extracted TICS given by load_TICS_dict
+    :type TICS_dict: {int: list} dict. The list is for taking into account repetitions of a TIC visualisation.
+
+    :returns: (boolean indicating existence, value of the dict entry if existing)
+    :rtype: (bool, (str, {str dict} list)) tuple
+    """
+    TIC_in_catalog = int(TIC_in_catalog)
+
+    if TIC_in_catalog in TICS_dict.keys():
+        return( (True, TICS_dict[TIC_in_catalog]))
+
+    else:
+        return ( (False, ("", []))) #we return False and default values
+
+def check_exists_other_ID(catalog_line):
+    """
+    Checks if the given line has another ID than a TIC
+
+    :param catalog_line: line of catalog currently being examined
+    :type catalog_line: dict
+
+    :returns: boolean indicating at least one other ID found, (name of mission ID, ID) dictionnary of found IDs
+    :rtype: (bool, dict) tuple
+    """
+    #Getting the fields corresponding to ids
+    list_of_ids = (os.getenv("OTHER_MISSIONS_IDS")).split(',')
+
+    #Dictionnary of found other ids
+    found_ids = {}
+    flag = False
+
+    #Checks for the existence of other ids
+    for id_mission_name in list_of_ids:
+        if catalog_line[id_mission_name] != None:
+            flag = flag or True
+            found_ids[id_mission_name] = catalog_line[id_mission_name]
```

Comment:
Those ids are not used anywhere in the code

Reply:
For now ^^